GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java

```diff
@@ -0,0 +1,2282 @@
+        expectedLogs.add("digraph {\n\tsubgraph cluster_0 {\n\t\t0 [label=\"9\"];\n\t\t0 -> 1;\n\t\t1 [label=\"10\"];\n\t\t0 -> 2;\n\t\t2 [label=\"11\"];\n\t\t2 -> 3;\n\t\t3 [label=\"12\"];\n\t\t0 -> 4;\n\t\t4 [label=\"13\"];\n\t\t4 -> 5;\n\t\t5 [label=\"14\"];\n\t\t4 -> 6;\n\t\t6 [label=\"15\"];\n\t\t6 -> 7;\n\t\t7 [label=\"16\"];\n\t}\n\tsubgraph cluster_8 {\n\t\t8 [label=\"2\"];\n\t\t8 -> 9;\n\t\t9 [label=\"19\"];\n\t\t8 -> 10;\n\t\t10 [label=\"3\"];\n\t\t10 -> 11;\n\t\t11 [label=\"4\"];\n\t}\n\tsubgraph cluster_12 {\n\t\t12 [label=\"5\"];\n\t\t12 -> 13;\n\t\t13 [label=\"6\"];\n\t\t12 -> 14;\n\t\t14 [label=\"7\"];\n\t\t14 -> 15;\n\t\t15 [label=\"8\"];\n\t}\n\tsubgraph cluster_16 {\n\t\t16 [label=\"17\"];\n\t\t16 -> 17;\n\t\t17 [label=\"18\"];\n\t}\n\t0 -> 8;\n\t8 -> 12;\n\t12 -> 16;\n}");
+        expectedLogs.add("digraph {\n\tsubgraph cluster_0 {\n\t\t0 [label=\"2\"];\n\t\t0 -> 1;\n\t\t1 [label=\"19\"];\n\t\t0 -> 2;\n\t\t2 [label=\"3\"];\n\t\t2 -> 3;\n\t\t3 [label=\"4\"];\n\t\t0 -> 4;\n\t\t4 [label=\"5\"];\n\t\t4 -> 5;\n\t\t5 [label=\"6\"];\n\t\t4 -> 6;\n\t\t6 [label=\"7\"];\n\t\t6 -> 7;\n\t\t7 [label=\"8\"];\n\t}\n\tsubgraph cluster_8 {\n\t\t8 [label=\"9\"];\n\t\t8 -> 9;\n\t\t9 [label=\"10\"];\n\t\t8 -> 10;\n\t\t10 [label=\"11\"];\n\t\t10 -> 11;\n\t\t11 [label=\"12\"];\n\t\t8 -> 12;\n\t\t12 [label=\"13\"];\n\t\t12 -> 13;\n\t\t13 [label=\"14\"];\n\t\t12 -> 14;\n\t\t14 [label=\"15\"];\n\t\t14 -> 15;\n\t\t15 [label=\"16\"];\n\t}\n\tsubgraph cluster_16 {\n\t\t16 [label=\"17\"];\n\t\t16 -> 17;\n\t\t17 [label=\"18\"];\n\t}\n\t0 -> 8;\n\t8 -> 16;\n}");
+        expectedLogs.add("digraph {\n\tsubgraph cluster_0 {\n\t\t0 [label=\"2\"];\n\t\t0 -> 1;\n\t\t1 [label=\"19\"];\n\t\t0 -> 2;\n\t\t2 [label=\"3\"];\n\t\t2 -> 3;\n\t\t3 [label=\"4\"];\n\t\t0 -> 4;\n\t\t4 [label=\"5\"];\n\t\t4 -> 5;\n\t\t5 [label=\"6\"];\n\t\t4 -> 6;\n\t\t6 [label=\"7\"];\n\t\t6 -> 7;\n\t\t7 [label=\"8\"];\n\t\t0 -> 8;\n\t\t8 [label=\"9\"];\n\t\t8 -> 9;\n\t\t9 [label=\"10\"];\n\t\t8 -> 10;\n\t\t10 [label=\"11\"];\n\t\t10 -> 11;\n\t\t11 [label=\"12\"];\n\t\t8 -> 12;\n\t\t12 [label=\"13\"];\n\t\t12 -> 13;\n\t\t13 [label=\"14\"];\n\t\t12 -> 14;\n\t\t14 [label=\"15\"];\n\t\t14 -> 15;\n\t\t15 [label=\"16\"];\n\t}\n\tsubgraph cluster_16 {\n\t\t16 [label=\"17\"];\n\t\t16 -> 17;\n\t\t17 [label=\"18\"];\n\t}\n\t0 -> 16;\n}");
+        assertEquals(1, deletedMin, "You deleted the wrong node/returned a wrong node as the deleted element! Expected<1> but got <" + deletedMin + ">");
```

Comment:
No need to specify expected in the message, assertEquals does that already.

Reply:
I was bored, so I thought of adding extra comments like the one from Artemis just for the sake of it. Is def unnecessary for sure